kb/2023-08-11/using_netapp_third_party_storage.md

```diff
@@ -0,0 +1,281 @@
+slug: install_netapp_trident_csi
+authors:
+  - name: Jeff Radick
+    title: Staff Software Engineer
+tags: [harvester]
+hide_table_of_contents: false
+---
+
+Instructions for installing the Netapp Astra Trident CSI driver into a Harvester cluster.
+
+This enables use of NetApp storage systems to store storage volumes usable by virtual machines running in Harvester.
+
+The NetApp storage will be an option in addition to the normal Longhorn storage; it will not replace Longhorn. Virtual machine images will still be stored using Longhorn.
+
+This has been tested with Harvester 1.2.0 and Trident v23.07.0.
+
+This procedure only works to access storage via iSCSI, not NFS.
+
+:::note
+3rd party storage classes (including those based on Trident) can only be used for non-boot volumes of Harvester VMs.
+:::
+
+# Detailed Instructions
+
+We are assuming that before beginning this procedure,
+a Harvester cluster and a NetApp ONTAP storage system
+are both installed and configured for use.
+
+Most of these steps can be performed on any system that has the `helm` and `kubectl` commands installed, and which has network connectivity to the management port of the Harvester cluster.  Let's call this your workstation.  Certain steps must be performed on one or more cluster nodes themselves.  The steps described below should be done on your workstation unless otherwise indicated.
+
+The last step (setting up multipathd) should be done on all nodes,
+after the Trident CSI has been installed.
+
+Certain parameters of your installation will require modification of details in the examples in the procedure given below. Those which you may wish to modify include:
+
+* The namespace.  `trident` is used as the namespace in the examples, but you may prefer to use another.
+* The name of the deployment. `mytrident` is used but you can change this to something else.
+* The management IP address of the ONTAP storage system
+* Login credentials (username and password) of the ONTAP storage system
+
+The procedure is as follows.
+
+1. Read the NetApp Astra Trident documentation:
+
+   * https://docs.netapp.com/us-en/trident/
+   * https://docs.netapp.com/us-en/trident/trident-get-started/kubernetes-deploy-operator.html
+   * https://docs.netapp.com/us-en/trident/trident-get-started/kubernetes-deploy-helm.html#deploy-the-trident-operator-and-install-astra-trident-using-helm
+
+   The simplest method is to install using Helm; that process is described here.
+
+1. Download the KubeConfig from the Harvester cluster.
+
+   * Open the web UI for your Harvester cluster
+   * In the lower left corner, click the "Support" link.  This will take you to a "Harvester Support" page.
+   * Click the button labeled "Download KubeConfig".  This will download a your cluster config in a file called "local.yaml" by default.
+   * Move this file to a convenient location and set your `KUBECONFIG` environment variable to the path of this file.
```

Comment:
```suggestion
1. Download the KubeConfig from the Harvester cluster.

   1. Open the web UI for your Harvester cluster.
   1. In the lower-left corner, select the **Support** link. This will take you to the **Harvester Support** page.
   1. Select "Download KubeConfig". This will download your cluster's config file, named `local.yaml` by default.
   1. Move this file to a convenient location and set your `KUBECONFIG` environment variable to the path of this file.
```